flax/nnx/nnx/bridge/wrappers.py

```diff
@@ -55,43 +56,52 @@ def _functional_constructor(*args: tp.Any, **kwargs: tp.Any) -> Functional[M]:
   return _functional_constructor
 
 
-class LinenWrapper(Module):
+def shaped_init(module: Module, *args, **kwargs):
+  """To trigger init of all `LinenToNNX` module variables and return a wholesome state."""
+  assert callable(module)
+  _ = module(*args, **kwargs)
+  return nnx.split(module)
```

Comment:
In this function we should leverage the fact that `Object._object__state._initializing` still exists and set if via something like

```python
def _set_initializing(initializing: bool):
  for _, value in graph.iter_graph(module):
    if isinstance(value, Object):
      value._object__state._initializing = initializing
```
and use the value of `_initializing` to choose between `init` and `apply` when calling the Linen Modules.

Reply:
Added `_set_initializing` to the LinenToNNX wrapper. 
Note that we can't do check on top level modules' `._object__state._initializing` because the top level module might be a pure NNX module with `._object__state._initializing` always False.

Reply:
I'd do something like this:

```python
def _set_initializing(module, initializing: bool): 
  for _, value in graph.iter_graph(module): 
    if isinstance(value, Object): 
      value._object__state._initializing = initializing
      
def shaped_init(module: Module, *args, **kwargs):
  """To trigger init of all `LinenToNNX` module variables and return a wholesome state."""
  module = graph.clone(module) # create a copy
  _set_initializing(module, True)
  assert callable(module)
  try:
    _ = module(*args, **kwargs)
  finally:
    _set_initializing(module, False)
  return nnx.split(module)
```

Reply:
Done, also renamed to `lazy_init` as discussed offline.